tests/vcs/bitbucket-spec.js

```diff
@@ -0,0 +1,277 @@
+    fetchStub = sandbox.stub()
+    revertFetchRewire = Bitbucket.__set__('fetch', fetchStub)
+
+    config = {
+      ci: {
+        gitUser: {
+          email: 'bamboo@my-domain.com',
+          name: 'bamboo'
+        },
+        provider: 'bamboo'
+      },
+      computed: {
+        vcs: {
+          auth: {
+            username: 'ci-user',
+            password: 'ci user password'
+          }
+        }
+      },
+      vcs: {
+        domain: 'api.bitbucket.org',
+        provider: 'bitbucket',
+        repository: {
+          name: 'repo',
+          owner: 'owner'
+        }
+      }
+    }
+
+    bitbucket = new Bitbucket(config)
+  })
+
+  afterEach(function () {
+    revertFetchRewire()
+    sandbox.restore()
+  })
+
+  it('should save the config', function () {
+    expect(bitbucket.config).to.be.eql(config)
+  })
+
+  it('should construct a 1.0 base URL', function () {
+    const url = 'https://ci-user:ci%20user%20password@api.bitbucket.org/1.0'
+    expect(bitbucket.baseUrl1).to.be.equal(url)
+  })
+
+  it('should construct a 2.0 base URL', function () {
+    const url = 'https://ci-user:ci%20user%20password@api.bitbucket.org/2.0'
+    expect(bitbucket.baseUrl2).to.be.equal(url)
+  })
+
+  describe('.addRemoteForPush()', function () {
+    let remoteName
+    beforeEach(function () {
+      return bitbucket.addRemoteForPush().then((name) => {
+        remoteName = name
+      })
+    })
+
+    it('should resolve with origin', function () {
+      expect(remoteName).to.be.equal('origin')
+    })
+  })
+
+  describe('.getPr()', function () {
+    let resolution, rejection, promise, fetchResolver
+    beforeEach(function () {
+      fetchResolver = {}
+      let fetchPromise = new Promise((resolve, reject) => {
+        fetchResolver.resolve = resolve
+        fetchResolver.reject = reject
+      })
+
+      fetchStub.returns(fetchPromise)
+
+      resolution = rejection = null
+      promise = bitbucket.getPr('5')
+        .then((resp) => {
+          resolution = resp
+          return resolution
+        })
+        .catch((err) => {
+          rejection = err
+          throw err
+        })
+    })
+
+    it('should call fetch with proper params', function () {
+      expect(fetchStub).to.have.been.calledWith(`${bitbucket.baseUrl2}/repositories/owner/repo/pullrequests/5`)
+    })
+
+    describe('when fetch resolves with success', function () {
+      let resp
+      beforeEach(function (done) {
+        resp = {ok: true, status: 200, json () {}}
+        let pr = {
+          id: 5,
+          description: 'This is a #fix#',
+          source: {
+            commit: {
+              hash: 'sha-1'
+            }
+          },
+          links: {
+            self: {
+              href: 'my-link-to-myself'
+            }
+          }
+        }
+        sandbox.stub(resp, 'json').returns(Promise.resolve(pr))
+
+        promise.then(() => {
+          done()
+        })
+
+        fetchResolver.resolve(resp)
```

Comment:
Similarly here, if you don't want to worry about the `done`, you can return the `promise.then(() => {})`, after this line of course :)

Reply:
I had an issue updating this, so I left it as is.